### DISCUSSION TITLE
Manage Retained Messages

### DISCUSSION THREAD

**Discussion by mauro-rogledi** [dotnet/MQTTnet#1430] [Q&A] [answered]
Where can I find an example for the new management of RetainedMessages.
I don't understand the logic of how to store messages
Thanks

**Comment by mauro-rogledi** [comment, marked as answer]
I found.
But I can't clear the discussion

**Reply by SeppPenner** [reply]
I did it. Do you mind sharing your solution? Maybe, I can add it to the docs under the upgrading section?

**Reply by mauro-rogledi** [reply]
I don't have a solution.
I added these events to my code and see when it was called
I'm sorry

            _mqttServer.LoadingRetainedMessageAsync += _mqttServer_LoadingRetainedMessageAsync;
            _mqttServer.RetainedMessageChangedAsync += _mqttServer_RetainedMessageChangedAsync;
            _mqttServer.RetainedMessagesClearedAsync += _mqttServer_RetainedMessagesClearedAsync;
